src/Mailgun/Mailgun.php

```diff
@@ -36,17 +43,79 @@ class Mailgun
     protected $apiKey;
 
     /**
-     * @param string|null $apiKey
-     * @param HttpClient  $httpClient
-     * @param string      $apiEndpoint
+     * @var HttpMethodsClient
+     */
+    private $httpClient;
+
+    /**
+     * @var ResponseSerializer
+     */
+    private $serializer;
+
+    /**
+     * @param string|null             $apiKey
+     * @param HttpClient|null         $httpClient
+     * @param string                  $apiEndpoint
+     * @param ResponseSerializer|null $serializer
      */
     public function __construct(
         $apiKey = null,
         HttpClient $httpClient = null,
-        $apiEndpoint = 'api.mailgun.net'
+        $apiEndpoint = 'api.mailgun.net',
+        ResponseSerializer $serializer = null
     ) {
         $this->apiKey = $apiKey;
+        $this->serializer = $serializer ?: new ObjectSerializer();
         $this->restClient = new RestClient($apiKey, $apiEndpoint, $httpClient);
+
+        $plugins = [
```

Comment:
So you would use a factory to create a PluginClient here and then i AbsractApi create a methods client?